ext/spl/spl_array.c

```diff
@@ -783,11 +780,18 @@ static inline HashTable* spl_array_get_debug_info(zend_object *obj) /* {{{ */
 		storage = &intern->array;
 		Z_TRY_ADDREF_P(storage);
 
-		base = obj->handlers == &spl_handler_ArrayIterator
+		const zend_class_entry *base_class_ce = instanceof_function(obj->ce, spl_ce_ArrayIterator)
```

Comment:
Given SPL's funkyness I'm not so sure changing this is a good idea, this call is also more expensive btw than the previous code.

Reply:
I was going to remove the dedicated object handlers as they are _identical_ to the ArrayObject ones, I'd also expect the base case to be fast as subclassing should be relatively rare

Reply:
Another reason is that I'd like to move the ArrayIterator into the iterator.c file in the long run

Reply:
Okay fair enough